### PR TITLE
Create production release

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
Creates a Github Release to trigger a production deploy. The release body just has the list of PRs which it includes for now. This code _should_ make it pretty easy to add more changelog notes and/or other files or artifacts with the release.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/155221

### Tests
Still haven't tested [action-gh-release](https://github.com/softprops/action-gh-release) in Public-Test-Repo.

To live-test this PR:

0) Merge this PR
1) Close a StagingDeployCash, and assuming https://github.com/Expensify/Expensify.cash/pull/1858 is working as expected, then the `production` job of the `deploy` workflow should be triggered.
2) It should then create a Github Release matching the tag from the `StagingDeployCash` that was closed.
3) That should trigger all the platform-specific deploy workflows (although the actual production deploy may be skipped for the time being, until we flip the switch).

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -27,13 +27,35 @@ jobs:
...
-      - name: Checkout production
-        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
         with:
-          ref: production
+          fetch-depth: 0
           token: ${{ secrets.OS_BOTIFY_TOKEN }}
 
+      - name: Checkout production branch
+        run: git checkout production
```
Comment (AndrewGable) [inline .github/workflows/deploy.yml:36]: Can you combine here with actions/checkout?

**Review by AndrewGable — APPROVED** [review]

**Comment by roryabraham** [comment]
finishReleaseCycle: https://github.com/Expensify/Expensify.cash/actions/runs/666269006
prod PR: https://github.com/Expensify/Expensify.cash/pull/1889
prod automerge: https://github.com/Expensify/Expensify.cash/actions/runs/666269990
create release: https://github.com/Expensify/Expensify.cash/actions/runs/666271840 ❌

**Comment by roryabraham** [comment]
Retesting after merging https://github.com/Expensify/Expensify.cash/pull/1894 ...

- finishReleaseCycle: https://github.com/Expensify/Expensify.cash/actions/runs/666300457
- prod PR: https://github.com/Expensify/Expensify.cash/pull/1897
- prod automerge: https://github.com/Expensify/Expensify.cash/actions/runs/666301879
- deploy production: https://github.com/Expensify/Expensify.cash/actions/runs/666303694 ❌ 

Failed at a later step this time!

**Comment by roryabraham** [comment]
Retesting after merging https://github.com/Expensify/Expensify.cash/pull/1901 ... 

- finishReleaseCycle: https://github.com/Expensify/Expensify.cash/actions/runs/666330964
- prod PR: https://github.com/Expensify/Expensify.cash/pull/1904
- prod automerge: https://github.com/Expensify/Expensify.cash/actions/runs/666332059
- deploy production: https://github.com/Expensify/Expensify.cash/actions/runs/666333977 ❌ 

Failed at a later step this time! Getting closer... We'll pick this back up tomorrow.

**Comment by roryabraham** [comment]
I think the last error we saw last night was caused because there were no existing releases. We could add code to handle that edge case, but instead I think we can just create a release using the UI.

**Comment by roryabraham** [comment]
Retesting after manually creating a couple releases:

- finishReleaseCycle: https://github.com/Expensify/Expensify.cash/actions/runs/668688064
- prod PR: https://github.com/Expensify/Expensify.cash/pull/1917
- prod automerge: https://github.com/Expensify/Expensify.cash/actions/runs/668689468
- deploy production: https://github.com/Expensify/Expensify.cash/actions/runs/668692438 ❌ 

Failed at a later step again!

**Comment by roryabraham** [comment]
image

That doesn't match the [docs](https://github.com/softprops/action-gh-release)!

image

😠 ... probably has to do with the commit hash we're pointing at

**Comment by roryabraham** [comment]
Yeah, we're pointing at `b7e450d`, and:

image

I'm just going to point at the latest commit

**Comment by roryabraham** [comment]
Retesting after merging https://github.com/Expensify/Expensify.cash/pull/1921 ... 

- finishReleaseCycle: https://github.com/Expensify/Expensify.cash/actions/runs/668819608
- prod PR: https://github.com/Expensify/Expensify.cash/pull/1924
- deploy production: https://github.com/Expensify/Expensify.cash/actions/runs/668823331
- Release created: https://github.com/Expensify/Expensify.cash/releases/tag/1.0.2-92

🎉 IT WORKED 🎉

**Pull request merged**